### ISSUE TITLE
Race condition inside Akka.Persistence MemoryAdapter

### ISSUE THREAD

**Issue by Aaronontheweb** [akkadotnet/akka.net#2104]
Seen this show up in the build logs several times now:

```
 MemoryEventAdapterSpec.EventAdapter_should_work_when_plugin_defines_no_adapter	
 
First failure:  	2092	#1300	ravengerua <yu… (1)	21 Jun 16 18:52
Akka.Actor.InvalidActorNameException : Actor name "p1" is not unique!
   at Akka.Actor.Internal.NormalChildrenContainer.Reserve(String name) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Actor\ChildrenContainer\Internal\NormalChildrenContainer.cs:line 51
   at Akka.Util.Internal.InterlockedSpin.Swap[T](T& reference, Func`2 updater) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Util\Internal\InterlockedSpin.cs:line 32
   at Akka.Actor.ActorCell.ReserveChild(String name) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Actor\ActorCell.Children.cs:line 116
   at Akka.Actor.ActorCell.MakeChild(Props props, String name, Boolean async, Boolean systemService) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Actor\ActorCell.Children.cs:line 362
   at Akka.Actor.ActorCell.AttachChild(Props props, Boolean isSystemService, String name) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Actor\ActorCell.Children.cs:line 35
   at Akka.Persistence.Tests.MemoryEventAdapterSpec.EventAdapter_should_work_when_plugin_defines_no_adapter() in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka.Persistence.Tests\MemoryEventAdapterSpec.cs:line 386
```

**Comment by Horusiath** [comment]
This doesn't look like anything related to persistence plugin itself.

**Comment by Aaronontheweb** [comment]
That's just where the exception is thrown - the code that's being used to add the children itself (whatever is calling this) is buggy.

**Comment by alexvaluyskiy** [comment]
@cconstantin is this bug is still actual? Or you have fixed it here?
https://github.com/akkadotnet/akka.net/pull/2245

**Comment by cconstantin** [comment]
@alexvaluyskiy yep, this was fixed by #2245.
